samcli/commands/exceptions.py

```diff
@@ -25,6 +28,42 @@ def __init__(self, message, wrapped_from=None):
         click.ClickException.__init__(self, message)
 
 
+class UnhandledException(click.ClickException):
+    """
+    Exception class to re-wrap any exception that is not a UserException.
+    Typically this means there is a bug in SAM CLI.
+    """
+
+    GH_BUG_REPORT_URL = "https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title={title}"
+    # NOTE (hawflau): actual exitcode is 1 to not break existing behavior. Only report 255 to telemetry
+    exit_code = 1
+
+    def __init__(self, command: str, exception: Exception) -> None:
+        self._command = command
+        self._exception = exception
+        self.__traceback__ = self._exception.__traceback__
+
+        click.ClickException.__init__(self, type(exception).__name__)
+
+    def show(self, file: t.Optional[t.IO] = None) -> None:
```

Comment:
This is a minor nit, but I think we usually import the specific types directly, instead of importing the whole typing 🤔